client/src/components/relatedProducts/ProductCard.jsx

```diff
@@ -1,27 +1,68 @@
 import React from 'react';
+import EachCard from './EachCard.jsx';
 
-class ProductCard extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      placehoder: []
+const ProductCard = (props) => {
+
+
+  let combine = props.detailInfo.reduce((map, value) => {
+    map[value.product_id] = value;
+    return map;
+  }, {});
+  let newArry = [];
+  for (let i = 0; i < props.info.length; i++) {
+    let v2 = combine[props.info[i].id]
+    if (v2) {
+      newArry.push({
+        id: v2.product_id,
+        results: v2.results,
+        category: props.info[i].category,
+        description: props.info[i].description,
+        defaultPrice: props.info[i].default_price
+      })
     }
   }
 
-  render() {
-    return (
-      <div className="productCard">
-
-        <div className="img">
-          <img className="cardImg" src="https://images.unsplash.com/photo-1501088430049-71c79fa3283e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=668&q=80"/>
-        </div>
-        <div className="cardCategory">Category</div>
-        <div className="expProductName">Expanded product name with extra text</div>
-        <div className="cardPrice">$80.00</div>
-        <div className="cardRating">*****</div>
-      </div>
-    )
-  }
+  console.log('HERE==', newArry)
+
+  return (
+    <div className="productCard">
+      {newArry.map((each, index) =>
+        <EachCard info={each} key={index} />)}
+    </div>
+  )
 }
 
+// class ProductCard extends React.Component {
```

Comment:
You have a few spots like this with commented out chunks of code? Do you still need these? If so, definitely keep it in! But it not, you could delete it for better optimization.